plugins/treasuredata/access_key.go

```diff
@@ -0,0 +1,68 @@
+package treasuredata
+
+import (
+	"context"
+	"os"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AccessKey,
+		DocsURL:       sdk.URL("https://docs.treasuredata.com/display/public/PD/TD+Toolbelt"),
+		ManagementURL: sdk.URL("https://console.treasuredata.com/app/mp/ak"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "APIKey used to authenticate to Treasure Data.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			importer.TryEnvVarPair(map[string]sdk.FieldName{
+				"TREASURE_DATA_API_KEY": fieldname.APIKey,
+			}),
+			TryTreasureDataConfigFile(os.Getenv("TREASURE_DATA_CONFIG_PATH")),
```

Comment:
If these envvars are unset, we're calling `importer.TryFile` with an empty string for the path. Now I believe that leads to the expected behavior because the operating system will return a not found error for the file because the file "" doesn't exist. I'm wondering if we should depend on this to happen at the operating system level, or if we should already abort trying to import when the envvar is unset.

@florisvdg do you forese any issues with the current approach?

Reply:
The `importer.TryFile` helper already ignores files that don't exist, so I don't think there's an issue there. We can look into ways to make this a bit more rigid in the framework, but that doesn't have to block this PR.

Reply:
Thank you for the quick check, Floris!